### PR TITLE
Fix metadata

### PR DESCRIPTION
**Pull request by Seth-Karlo**

### CODE REVIEW

File: illumina_somaticVariants.pm [hunk 1]
```diff
@@ -34,8 +34,11 @@ sub parseSamples {
         if($opt{SET_HAS_METADATA} eq "yes"){
             print "Reading Tumour/Reference names from metadata\n\n";
             my $metadata = metadataParse($opt{OUTPUT_DIR});
-            push(@{$somatic_samples{$cpct_name}{"ref"}}, $metadata->{'ref_sample'});
-            push(@{$somatic_samples{$cpct_name}{"tumor"}}, $metadata->{'tumor_sample'});
+            if($sample =~ $metadata->{'ref_sample'}){
+              push(@{$somatic_samples{$cpct_name}{"ref"}}, $metadata->{'ref_sample'});
+            } elsif ($sample =~ $metadata->{'tumor_sample'}){
+              push(@{$somatic_samples{$cpct_name}{"tumor"}}, $metadata->{'tumor_sample'});
+            }
```
Comment (sambrightman) [inline illumina_somaticVariants.pm]: If we are using metadata and it does not match the sample name, I'd probably expect an error/warning in an `else` clause.

**Comment by sambrightman** [comment]
Prefer to see such commits (edits/corrections of a single logical change) rebased into a single commit.

**Comment by Seth-Karlo** [comment]
@sambrightman Fixed your comments

File: illumina_somaticVariants.pm [hunk 2]
```diff
@@ -34,8 +34,13 @@ sub parseSamples {
...
             print "Reading Tumour/Reference names from metadata\n\n";
             my $metadata = metadataParse($opt{OUTPUT_DIR});
-            push(@{$somatic_samples{$cpct_name}{"ref"}}, $metadata->{'ref_sample'});
-            push(@{$somatic_samples{$cpct_name}{"tumor"}}, $metadata->{'tumor_sample'});
+            if($sample =~ $metadata->{'ref_sample'}){
+              push(@{$somatic_samples{$cpct_name}{"ref"}}, $metadata->{'ref_sample'});
+            } elsif ($sample =~ $metadata->{'tumor_sample'}){
+              push(@{$somatic_samples{$cpct_name}{"tumor"}}, $metadata->{'tumor_sample'});
+            } else {
+              print "WARNING: Somatic variants are set to yes but metadata does not match sample names";
```
Comment (sambrightman) [inline illumina_somaticVariants.pm]: For clarity, maybe something like `"Metadata is set to yes but \"$sample\" does not match a metadata sample name"`? What if the metadata file exists but does not contain the right keys?

Comment (Seth-Karlo) [inline illumina_somaticVariants.pm]: As we'll be the only ones running it, I think it'll be clear enough. The important thing is the functionality, this thing will never be a glaring example of excellent code.

Comment (sambrightman) [inline illumina_somaticVariants.pm]: In that case I'd say we can probably dispense with code reviews.

**Pull request merged**